### PR TITLE
Waveforms

### PR DESCRIPTION
**Pull request by 44100hertz**
- [x] Drop down menu
- [x] Triangle wave
- [x] Hump sine
- [x] Show active wave
- [x] Fix preset
- [x] Add README documentation
- [x] Add Changelog documentation
- [x] version 0.2.0

### CODE REVIEW

**Comment by neonfuz** [comment]
When you load a file the waves don't get properly restored

**Comment by 44100hertz** [comment]
Correct it did not load instruments at all. Bug fixed.

**Comment by neonfuz** [comment]
Ah I see. I think before merge you can squash these commits, its best to have logical commits that make sense when reading the whole history. There's a squash and merge feature on github too if it makes sense to make it all a single commit (but it doesn't always)

**Comment by neonfuz** [comment]
I'm gonna try to implement the drop down box

**Comment by neonfuz** [comment]
Made it a tad wider to compensate for the arrow. Apparently it's possible but non trivial to word wrap select boxes. I say this is fine

**Comment by neonfuz** [comment]
heh, interesting, I can't request you do a code review (probably because you're OP) even though I pushed changes

File: src/lib/Oscilloscope.svelte
Comment (neonfuz) [inline src/lib/Oscilloscope.svelte:7]: I'm interested in the rationale. Oscilloscopes were upside down?

File: src/lib/OscilloscopePanel.svelte
Comment (neonfuz) [inline src/lib/OscilloscopePanel.svelte:19]: first sample was invalid / bad?

File: src/lib/audio/instrument.ts
Comment (neonfuz) [inline src/lib/audio/instrument.ts:7]: Remember to remove dev tags before release!

File: src/lib/audio/migration.ts
Comment (neonfuz) [inline src/lib/audio/migration.ts:28]: did you mean to leave this?

**Review by neonfuz — CHANGES_REQUESTED** [review]
- remove dev tag from instrument names (find & replace project wide)
- remove extra console.log

File: src/lib/audio/migration.ts
Comment (44100hertz) [reply src/lib/audio/migration.ts:28]: Will remove

**Comment by 44100hertz** [comment]
image
Instead of a drop down menu, I think a little button selector like this is ideal. I can make it happen.

**Comment by 44100hertz** [comment]
NOT ready for merge yet. 1 or 2 more features to add.

**Comment by neonfuz** [comment]
Nice I like the new wave selector!

I did notice that some of the waves appear upside down, or starting at a different point than the preview icons:

2022-09-07T13:32:46:-05:00_
2022-09-07T13:32:50:-05:00_
2022-09-07T13:33:13:-05:00_
2022-09-07T13:33:17:-05:00_

File: src/lib/Oscilloscope.svelte
Comment (neonfuz) [reply src/lib/Oscilloscope.svelte:7]: Is this why the graphs are upside down? @44100hertz

File: src/lib/WaveIcon.svelte
Comment (neonfuz) [inline src/lib/WaveIcon.svelte:15]: waveIcon class is pointless here because you're not using it in CSS. Either:
- remove it here and rely on svelte style scoping, or
- use it in CSS like .waveIcon.sine {...}

File: src/lib/WaveIcon.svelte
Comment (neonfuz) [inline src/lib/WaveIcon.svelte:9]: Remove width and height, they do nothing here because you override them later in css.

File: src/lib/audio/instrument.ts
Comment (neonfuz) [reply src/lib/audio/instrument.ts:7]: Adding this to the review

File: src/lib/audio/instrument.ts
Comment (neonfuz) [inline src/lib/audio/instrument.ts:34]: change

File: src/lib/audio/instruments/Synth Bass.json
Comment (neonfuz) [inline src/lib/audio/instruments/Synth Bass.json:2]: change

**Review by neonfuz — CHANGES_REQUESTED** [review]

File: src/lib/Oscilloscope.svelte
Comment (44100hertz) [reply src/lib/Oscilloscope.svelte:7]: The SVG coordinate space is that Y goes from top to bottom, which is the opposite of how audio is traditionally displayed.

File: src/lib/OscilloscopePanel.svelte
Comment (44100hertz) [reply src/lib/OscilloscopePanel.svelte:19]: Yea the envelope takes 1 sample to kick in. I could try to fix that in the envelope generator, instead.

**Comment by 44100hertz** [comment]
NOTE: latest deploy is a hotfix, it is not versioning properly. This should be fixed before merging.

File: src/lib/WaveIcon.svelte
Comment (44100hertz) [reply src/lib/WaveIcon.svelte:9]: done

File: src/lib/audio/instrument.ts
Comment (44100hertz) [reply src/lib/audio/instrument.ts]: we handle versions differently now...

**Comment by 44100hertz** [comment]
@neonfuz can you merge this?

File: .gitignore
Comment (neonfuz) [inline .gitignore:9]: Pretty sure this was added by accident. The project no longer generates this bundle.

**Review by neonfuz — CHANGES_REQUESTED** [review]
Pretty sure you accidentally added old files to .gitignore

You should check next time by deleting the file and making sure it was actually being generated by the project before adding it to .gitignore

File: .gitignore
Comment (44100hertz) [reply .gitignore]: fixd

**Review by neonfuz — APPROVED** [review]

**Comment by neonfuz** [comment]
I would merge but it says we have conflicts, if you resolve them I will merge

**Comment by 44100hertz** [comment]
WTF I'm rebasing wrong or something. Don't merge this

**Pull request merged**
